src/DnsHelperUI/MainForm.cs

```diff
@@ -17,7 +17,7 @@ public partial class MainForm : Form
         /// <summary>
         /// The currently selected Network interface card
         /// </summary>
-        public string CurrentNic { get { return btnSelectedNic.Text; } set { btnSelectedNic.Text = value; } }
+        public string CurrentNic; //{ get { return btnSelectedNic.Text; } set { btnSelectedNic.Text = value; } }
```

Comment:
Please leave `CurrentNic` as a public property and consider having it trigger the DNS refresh logic when the value is updated. Would make the structure less error prone, example: 
```
private string _currentNic;
public string CurrentNic
{
    get { return _currentNic; }
    set
    {
        _currentNic = value;

        // Update the DNS values to the currently applied ones
        var dnsServers = RefreshDNSValues();
        SetDnsTextBoxValues(dnsServers != null && dnsServers.Length > 0 ? dnsServers[0] : string.Empty,
                            dnsServers != null && dnsServers.Length > 1 ? dnsServers[1] : string.Empty);
    }
}
```